### ISSUE TITLE
rtrim($url, '#') or not - that is the question

### ISSUE THREAD

**Issue by massadm** [swaggest/php-json-schema#35]
Hello again!

Issue related to `Swaggest\JsonSchema\RemoteRef\Preloaded` but...

If the "$schema" and "id" has "#"  at the end, then someone must need them?

`{ "id": "https://some-domain/Foo/Bar#" }` and `{ "id": "https://some-domain/Foo/Bar" }` is the same schema id?

Currently `Swaggest\JsonSchema\RefResolver::updateResolutionScope` do `rtrim($id, '#')` whereas `Swaggest\JsonSchema\RemoteRef\Preloaded::getSchemaData` and `Swaggest\JsonSchema\RemoteRef\Preloaded::setSchemaData` do not.

This leads to uncoordinated behavior, eg:

```
$options = new Context();
$provider = new Preloaded();
$provider->setSchemaData('https://some-domain/Baz/Bar#',(object)['type'=>'integer']);
$options->setRemoteRefProvider($provider);
$schemaData = (object)[
	'id' => 'https://some-domain/Foo/Bar#',
	'anyOf' => [
		(object)['$ref' => '/Baz/Bar#']
	]
];
$schema = Schema::import($schemaData, $options);

```
`json_encode($schema)`
```
{"id":"https://some-domain/Foo/Bar#","anyOf":[{"not":{}}]}
```
This is because id-based key (without "#" due to `rtrim()`) is missed in `$storage`!

`$provider->setSchemaData('https://some-domain/Baz/Bar',(object)['type'=>'integer'])` (first argument trimmed) is OK

```
{"id":"https://some-domain/Foo/Bar#","anyOf":[{"type":"integer"}]}
```

What behavior do you think is right?

**Comment by massadm** [comment]
Related information.

http://json-schema.org/draft-07/schema#/$id
```
{
    "type": "string",
    "format": "uri-reference"
}
```

URI vocabulary: ["URI" vs. "URI reference"](https://github.com/json-schema/json-schema/issues/81)

**Comment by vearutop** [comment]
Indeed references to `http://some-where/resource` and `http://some-where/resource#` are identical. Thanks again for thorough debug and analysis, send a PR next time 👍

**Comment by vearutop** [comment]
Fixed in `v0.11.3`.
